contracts/auction/RevenuePoolV2.sol

```diff
@@ -0,0 +1,116 @@
+pragma solidity ^0.4.24;
+
+import "@evolutionland/common/contracts/interfaces/ISettingsRegistry.sol";
+import "@evolutionland/common/contracts/interfaces/ERC223ReceivingContract.sol";
+import "@evolutionland/common/contracts/interfaces/ERC223.sol";
+import "@evolutionland/common/contracts/interfaces/IUserPoints.sol";
+import "@evolutionland/common/contracts/DSAuth.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./AuctionSettingIds.sol";
+import "./interfaces/IGovernorPool.sol";
+
+/**
+ * @title RevenuePool
+ * difference between LandResourceV1:
+     change DividendPool into governorPool for reward
+ */
+
+// Use proxy mode
+contract RevenuePoolV2 is DSAuth, ERC223ReceivingContract, AuctionSettingIds {
+
+    bool private singletonLock = false;
+
+//    // 10%
+//    address public pointsRewardPool;
+//    // 30%
+//    address public contributionIncentivePool;
+//    // 30%
+//    address public dividendsPool;
+//    // 30%
+//    address public devPool;
+
+    ISettingsRegistry public registry;
+
+    // claimedToken event
+    event ClaimedTokens(address indexed token, address indexed owner, uint amount);
+
+    /*
+     *  Modifiers
+     */
+    modifier singletonLockCall() {
+        require(!singletonLock, "Only can call once");
+        _;
+        singletonLock = true;
+    }
+
+    function initializeContract(address _registry) public singletonLockCall {
+        owner = msg.sender;
+        emit LogSetOwner(msg.sender);
+
+        registry = ISettingsRegistry(_registry);
+    }
+
+    function tokenFallback(address _from, uint256 _value, bytes _data) public {
+
+        address ring = registry.addressOf(SettingIds.CONTRACT_RING_ERC20_TOKEN);
+        address userPoints = registry.addressOf(SettingIds.CONTRACT_USER_POINTS);
+
+        if(msg.sender == ring) {
+            address buyer = bytesToAddress(_data);
+            // should same with trading reward percentage in settleToken;
+
+            IUserPoints(userPoints).addPoints(buyer, _value / 1000);
+        }
+    }
+
+
+    function settleToken(address _tokenAddress) public {
+        uint balance = ERC20(_tokenAddress).balanceOf(address(this));
+
+        // to save gas when playing
+        if (balance > 10) {
+            address pointsRewardPool = registry.addressOf(AuctionSettingIds.CONTRACT_POINTS_REWARD_POOL);
+            address contributionIncentivePool = registry.addressOf(AuctionSettingIds.CONTRACT_CONTRIBUTION_INCENTIVE_POOL);
+            address governorPool = registry.addressOf(AuctionSettingIds.CONTRACT_GOVERNOR_POOL);
+            address devPool = registry.addressOf(AuctionSettingIds.CONTRACT_DEV_POOL);
```

Comment:
Suggest to continue to use CONTRACT_DIVIDENDS_POOL to save & simplify upgration and gas.

Reply:
ok fix it.